btree/btree.c

```diff
@@ -1072,7 +1073,8 @@ static void node_del  (const struct nd *node, int idx, struct m0_be_tx *tx);
 static void node_set_level  (const struct nd *node, uint8_t new_level);
 static void node_move (struct nd *src, struct nd *tgt,
 		       enum dir dir, int nr, struct m0_be_tx *tx);
-static void mem_update(void);
+static void mem_update(struct node_op *op, struct m0_be_tx *tx, void *ptr,
```

Comment:
I see you had introduced this function. I am not sure why we need this one. Also you have added this function amongst the node_* functions. If this is specific to mem_* operations then this function needs to be moved to the proper location. Can you also please explain the intention behind the existence of this function ?

Reply:
The reason I have added this mem_update() is that it will call our m0_be_tx_capture() function to write data to segment.

Reply:
If the node functions are the ones which change the on-disk structures then they are the ones which need the tx pointer and also it is their responsibility to capture their modifications  in the tx. Apart from the node functions (including tree_create and tree_destroy) if any other functions make changes to on-disk structures then we need to discuss about them to see if we need to have some more node_* functions too. Currently I do not feel mem_update is necessary but we can talk about this and discuss.

Reply:
Removed mem_update and added commented line where m0_be_tx_capture() is expected to be called.